enterprise/server/byte_stream_server_proxy/byte_stream_server_proxy.go

```diff
@@ -65,33 +148,67 @@ func (s *ByteStreamServerProxy) Read(req *bspb.ReadRequest, stream bspb.ByteStre
 }
 
 func (s *ByteStreamServerProxy) Write(stream bspb.ByteStream_WriteServer) error {
-	remoteStream, err := s.remoteCache.Write(stream.Context())
+	local, err := s.local.Write(stream.Context())
+	if err != nil {
+		local = nil
+	}
+	remote, err := s.remote.Write(stream.Context())
 	if err != nil {
 		return err
 	}
+
 	for {
 		req, err := stream.Recv()
 		if err != nil {
 			return err
 		}
-		writeDone := req.GetFinishWrite()
-		if err := remoteStream.Send(req); err != nil {
+
+		// Send to the local ByteStreamServer (if it hasn't errored)
+		localDone := req.GetFinishWrite()
+		if local != nil {
+			if err := local.Send(req); err != nil {
+				if err == io.EOF {
+					localDone = true
+				} else {
+					log.Infof("error writing to local bytestream server: %s", err)
+				}
+				local = nil
```

Comment:
what's the reason for nilling out local here?

Reply:
to prevent sending addition frames to the local server if it returned an error